integration/compiler-mjs-cjs-output-test.ts

```diff
@@ -0,0 +1,76 @@
+import { test, expect } from "@playwright/test";
+import * as fs from "node:fs";
+import * as path from "node:path";
+
+import { createFixtureProject, js, json } from "./helpers/create-fixture.js";
+
+test.describe("", () => {
+  for (let [serverModuleExt, serverModuleFormat, exportStatement] of [
+    ["mjs", "esm", "export {"],
+    ["cjs", "cjs", "module.exports ="],
+  ]) {
+    test(`can write .${serverModuleExt} server output module`, async () => {
+      let projectDir = await createFixtureProject({
+        files: {
+          // Ensure the config is valid ESM
+          "remix.config.js": js`
+            export default {
+              serverModuleFormat: "${serverModuleFormat}",
+              serverBuildPath: "build/index.${serverModuleExt}",
+            };
+          `,
+          "package.json": json({
+            name: "remix-template-remix",
+            private: true,
+            sideEffects: false,
+            type: "module",
+            dependencies: {
+              "@remix-run/node": "0.0.0-local-version",
+              "@remix-run/react": "0.0.0-local-version",
+              "@remix-run/serve": "0.0.0-local-version",
+              isbot: "0.0.0-local-version",
+              react: "0.0.0-local-version",
+              "react-dom": "0.0.0-local-version",
+            },
+            devDependencies: {
+              "@remix-run/dev": "0.0.0-local-version",
+              "@types/react": "0.0.0-local-version",
+              "@types/react-dom": "0.0.0-local-version",
+              typescript: "0.0.0-local-version",
+            },
+            engines: {
+              node: ">=18.0.0",
+            },
+          }),
```

Comment:
`node-template`'s `package.json` already has `"type": "module",`, so is there a specific reason why we need a custom `package.json`?

```suggestion
```

Reply:
I was just using what was already there in this integration test. I agree, but that increases the scope of this PR. Is that alright with you?

Reply:
Ah yeah now I see you just copy/pasted the other test and made some changes

Imo we can just remove these files as they're not really necessary for these tests 🤷‍♂️